### PR TITLE
fix: Config not loading environment variables for config

### LINKED ISSUE #2135: wr.config.s3_endpoint_url is not loaded
### Describe the bug

Env variable `WR_S3_ENDPOINT_URL` is not loaded into the `s3_endpoint_url` variable in `wr.config`. 

Setting it manually by `wr.config.s3_endpoint_url = os.environ['WR_S3_ENDPOINT_URL']` solves the problem.

### How to Reproduce

```
# In[1]:
import pandas as pd
import awswrangler as wr

# In[2]:
wr.__version__

# Out[2]:
'2.20.0'

# In[3]:
wr.config.s3_endpoint_url

# Out[3]:


# In[4]:
wr.s3.does_object_exist("path_to_file")

# Out[4]:
---------------------------------------------------------------------------
ClientError                               Traceback (most recent call last)
...
ClientError: An error occurred (403) when calling the HeadObject operation: Forbidden
```

### Expected behavior

```
# In[1]:
import pandas as pd
import awswrangler as wr

# In[2]:
wr.__version__

# Out[2]:
'2.20.0'

# In[3]:
wr.config.s3_endpoint_url

# Out[3]:
# returns endpoint url

# In[4]:
wr.s3.does_object_exist("path_to_file")

# Out[4]:
True
```

### Your project

_No response_

### Screenshots

_No response_

### OS

Linux

### Python version

3.9.10

### AWS SDK for pandas version

2.20.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Feature or Bugfix
- Bugfix

### Detail
- Fix config not loading environment variables for config

### Issues
- Fixes #2135

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 946bb1abc75eb91ce90a7933c06e4c48abd40861
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4cb8b48c-ea6c-4270-a5f4-7da890381176%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kstajer — APPROVED** [review]

**Review by kukushking — APPROVED** [review]

File: tests/test_config.py [hunk 1]
```diff
@@ -258,3 +258,12 @@ def test_athena_wait_delay_config_override(wr: ModuleType, glue_database: str) -
         mock_wait_query.assert_called_once()
 
         assert mock_wait_query.call_args[1]["athena_query_wait_polling_delay"] == polling_delay_argument
+
+
+def test_load_from_env_variable(wr: ModuleType) -> None:
+    s3_endpoint_url = "s3.us-east-1.amazonaws.com"
```
Comment (jaidisido) [inline tests/test_config.py:264]: Nitpick, but perhaps we should avoid making this test specific to `us-east-1`? Either a regex or we choose a different env variable to test?

**Review by jaidisido — APPROVED** [review]
LGTM, left a minor comment

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: ec3359a7e638e0694c8c3d83c983fe0e8900412a
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=71897232-23e9-4078-a3b2-596e8e557e63%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**